server/graphql/v2/mutation/VendorMutations.ts

```diff
@@ -215,6 +215,66 @@ const vendorMutations = {
       return true;
     },
   },
+  convertOrganizationToVendor: {
+    type: new GraphQLNonNull(GraphQLVendor),
+    description: 'Convert an organization to a vendor',
+    args: {
+      organization: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'Reference to the organization to convert',
+      },
+      host: {
+        type: new GraphQLNonNull(GraphQLAccountReferenceInput),
+        description: 'Reference to the host that will hold the vendor',
+      },
+    },
+    resolve: async (_, args, req) => {
+      checkRemoteUserCanUseHost(req);
+
+      const organization = await fetchAccountWithReference(args.organization, {
+        loaders: req.loaders,
+        throwIfMissing: true,
+      });
+      assert(organization.type === CollectiveType.ORGANIZATION, new ValidationFailed('Account is not an Organization'));
+      assert(!organization.HostCollectiveId, new ValidationFailed('Organization is hosted by another collective'));
+
+      const host = await fetchAccountWithReference(args.host, { loaders: req.loaders, throwIfMissing: true });
+      assert(
+        req.remoteUser.isAdminOfCollective(host),
+        new Unauthorized("You're not authorized to convert this organization"),
+      );
+
+      const transactions = await models.Transaction.findAll({
+        where: {
+          FromCollectiveId: organization.id,
+        },
+      });
+      assert(
+        transactions.every(t => t.HostCollectiveId === host.id),
+        new ValidationFailed('Cannot convert an organization with transactions to another fiscal-host'),
+      );
+
+      const mapId = ar => ar.map(({ id }) => id);
+      const hostAdmins = await host.getAdminUsers().then(mapId);
+      const organizationAdmins = await organization.getAdminUsers().then(mapId);
+      const alienAdmins = difference(organizationAdmins, hostAdmins);
+      assert(
+        alienAdmins.length === 0,
+        new ValidationFailed(`Cannot convert an organization with admins that are not admins of the new host`),
+      );
+
+      const vendorData = {
+        type: CollectiveType.VENDOR,
+        slug: `${host.id}-${organization.slug}-${uuid().substr(0, 8)}`,
+        CreatedByUserId: req.remoteUser.id,
+        isActive: false,
+        ParentCollectiveId: host.id,
+        data: organization.data || {},
+      };
+      vendorData.data['originalOrganizationProps'] = pick(organization.toJSON(), Object.keys(vendorData));
+      return organization.update(vendorData);
```

Comment:
I would also delete all the admins of the organization, something like:

```suggestion
      await organization.update(vendorData);
      await models.Member.destroy({ where: { CollectiveId: organization.id } });
      return organization;
```